src/gui/guiFormSpecMenu.cpp

```diff
@@ -3767,14 +3772,46 @@ void GUIFormSpecMenu::updateSelectedItem()
 			if (item.empty())
 				continue;
 
-			// Grab selected item from the crafting result list
-			m_selected_item = new GUIInventoryList::ItemSpec;
-			m_selected_item->inventoryloc = e->getInventoryloc();
-			m_selected_item->listname = "craftresult";
-			m_selected_item->i = 0;
-			m_selected_item->slotsize = e->getSlotSize();
-			m_selected_amount = item.count;
-			m_selected_dragging = false;
+			GUIInventoryList::ItemSpec s = GUIInventoryList::ItemSpec();
+			s.inventoryloc = e->getInventoryloc();
+			s.listname = "craftresult";
+			s.i = 0;
+			s.slotsize = e->getSlotSize();
+
+			if (m_shift_craft) {
```

Comment:
Suggestion: Invert this condition and use its code block with an early `break;` to reduce indent levels.
Result: Hopefully an easier to follow code structure.